src/main/java/com/microsoft/azure/spring/data/cosmosdb/common/TelemetrySender.java

```diff
@@ -33,30 +34,45 @@
 
     private static final ObjectMapper MAPPER = new ObjectMapper();
 
-    private void executeRequest(final TelemetryEventData eventData) {
+    private static final int RETRY_LIMIT = 3; // Align the retry times with sdk
+
+    private ResponseEntity<String> executeRequest(final TelemetryEventData eventData) {
         final HttpHeaders headers = new HttpHeaders();
-        ResponseEntity<String> response = null;
 
         headers.add(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON.toString());
 
         try {
             final RestTemplate restTemplate = new RestTemplate();
             final HttpEntity<String> body = new HttpEntity<>(MAPPER.writeValueAsString(eventData), headers);
 
-            response = restTemplate.exchange(TELEMETRY_TARGET_URL, HttpMethod.POST, body, String.class);
+            return restTemplate.exchange(TELEMETRY_TARGET_URL, HttpMethod.POST, body, String.class);
         } catch (JsonProcessingException | HttpClientErrorException ignore) {
             log.warn("Failed to exchange telemetry request, {}.", ignore.getMessage());
         }
 
+        return null;
+    }
+
+    private void sendTelemetryData(@NonNull TelemetryEventData eventData) {
+        ResponseEntity<String> response = null;
+
+        for (int i = 0; i < RETRY_LIMIT; i++) {
```

Comment:
You should have some kind of retry policy instead of retry immediately after failure.

Reply:
Thanks for point out this. As we discussed, there will be another package for this, and it will take care of this carefully.